modules/social_features/social_event/modules/social_event_invite/src/SocialEventInviteConfigOverride.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace Drupal\social_event_invite;
+
+use Drupal\Component\Utility\EmailValidatorInterface;
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Config\ConfigFactoryOverrideInterface;
+use Drupal\Core\Config\StorageInterface;
+use Drupal\Core\Database\Connection;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Provides an overridden block for Settings Tray testing.
+ *
+ * @package Drupal\social_event_invite
+ */
+class SocialEventInviteConfigOverride implements ConfigFactoryOverrideInterface {
+
+  /**
+   * The current request.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * Email validator.
+   *
+   * @var \Drupal\Component\Utility\EmailValidatorInterface
+   */
+  protected $emailValidator;
+
+  /**
+   * The current active database's master connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * Constructs the configuration override.
+   *
+   * @param \Symfony\Component\HttpFoundation\RequestStack $request_stack
+   *   The request stack.
+   * @param \Drupal\Component\Utility\EmailValidatorInterface $email_validator
+   *   The email validator.
+   * @param \Drupal\Core\Database\Connection $database
+   *   The current active database's master connection.
+   */
+  public function __construct(
+    RequestStack $request_stack,
+    EmailValidatorInterface $email_validator,
+    Connection $database
+  ) {
+    $this->requestStack = $request_stack;
+    $this->emailValidator = $email_validator;
+    $this->database = $database;
+  }
+
+  /**
+   * Load overrides.
+   */
+  public function loadOverrides($names) {
+    $overrides = [];
+
+    $config_name = 'user.settings';
+
+    if (in_array($config_name, $names, TRUE)) {
+      $request = $this->requestStack->getCurrentRequest();
+
+      $invitee_mail = $request->query->get('invitee_mail', '');
+      $destination = $request->query->get('destination', '');
+
+      $is_valid = $this->validateInviteData($invitee_mail, $destination);
+
+      if ($is_valid) {
+        $overrides[$config_name]['verify_mail'] = FALSE;
+      }
+    }
+
+    return $overrides;
+  }
+
+  /**
+   * Validate invite data.
+   *
+   * @param string $invitee_mail
+   *   Encoded email address of invited user.
+   * @param string $destination
+   *   The url of invited group.
+   *
+   * @return bool
+   *   TRUE if invited data is valid.
+   */
+  public function validateInviteData($invitee_mail, $destination) {
+
+    if (empty($invitee_mail) || empty($destination)) {
+      return FALSE;
+    }
+
```

Comment:
Add more comments about what and why we do here.

Reply:
done